From 6ad45d7a2f5a45e5857d77e53ee3f0987fae1872 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Wed, 22 Feb 2012 14:12:36 -0200
Subject: [RHEL6 qemu-kvm PATCH 1/3] scsi: fix fw path

RH-Author: Paolo Bonzini <pbonzini@redhat.com>
Message-id: <1329919979-20948-80-git-send-email-pbonzini@redhat.com>
Patchwork-id: 37558
O-Subject: [RHEL 6.3 qemu-kvm PATCH v2 079/102] scsi: fix fw path
Bugzilla: 782029
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Orit Wasserman <owasserm@redhat.com>
RH-Acked-by: Gerd Hoffmann <kraxel@redhat.com>

The pre-1.0 firmware path for SCSI devices already included the LUN
using the suffix argument to add_boot_device_path.  I missed that when
making channel and LUN customizable.  Avoid that it is included twice, and
convert the colons to commas for consistency with other kinds of devices

Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
Signed-off-by: Kevin Wolf <kwolf@redhat.com>
(cherry picked from 7082826eb41887da7795b3135eaeda1726647987)

Conflicts:
	hw/scsi-disk.c (trivial)
---
 hw/scsi-bus.c  |    2 +-
 hw/scsi-disk.c |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 hw/scsi-bus.c  |    2 +-
 hw/scsi-disk.c |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/hw/scsi-bus.c b/hw/scsi-bus.c
index 63e8058..49c1bb7 100644
--- a/hw/scsi-bus.c
+++ b/hw/scsi-bus.c
@@ -1448,7 +1448,7 @@ static char *scsibus_get_fw_dev_path(DeviceState *dev)
     SCSIDevice *d = DO_UPCAST(SCSIDevice, qdev, dev);
     char path[100];
 
-    snprintf(path, sizeof(path), "%s@%d:%d:%d", qdev_fw_name(dev),
+    snprintf(path, sizeof(path), "%s@%d,%d,%d", qdev_fw_name(dev),
              d->channel, d->id, d->lun);
 
     return strdup(path);
diff --git a/hw/scsi-disk.c b/hw/scsi-disk.c
index 738b8a8..e32f6bd 100644
--- a/hw/scsi-disk.c
+++ b/hw/scsi-disk.c
@@ -1644,7 +1644,7 @@ static int scsi_initfn(SCSIDevice *dev)
     s->qdev.conf.bs->buffer_alignment = s->qdev.blocksize;
 
     bdrv_iostatus_enable(s->qdev.conf.bs);
-    add_boot_device_path(s->qdev.conf.bootindex, &dev->qdev, ",0");
+    add_boot_device_path(s->qdev.conf.bootindex, &dev->qdev, NULL);
     return 0;
 }
 
-- 
1.7.3.2

